### DISCUSSION TITLE
multiple styles for one element?

### DISCUSSION THREAD

**Discussion by trebahl** [dotnet/wpf#10088] [General]
I wish it were possible to apply a list of styles (typically applying to different properties to make sense) to an element, instead of a single style. At the moment, if we want to compose a number of styles, we need to base them upon one another which can be a pain.

In a somewhat related fashion, why is FrameworkElement.Triggers restricted to event triggers? (This is somewhat related because I tried filling it with the Triggers from the styles I wanted to apply and got shot down).

**Comment by miloush** [comment]
As long as you have clear rules on merging the styles in your mind, you could probably write a markup extension to produce a merged style for you.

What do you mean by PropertyTrigger? There is no `PropertyTrigger` class as far as I can tell.

**Reply by trebahl** [reply]
I mean the simple Trigger class, that gets triggered when a dependency property is set.

**Reply by trebahl** [reply]
In my case it would be merging by code. Is it advisable to create one style object per control instead of sharing a style among different controls of the same class? I'm wary of doing that since everything seems designed to encouraged the sharing.

If it's not a problem, then I can live with that, even though it feels needlessly complicated.

**Reply by miloush** [reply]
I see, there is a proposal to remove that limitation #10048.

Depends on how many controls you have, merging also costs something, so it's a memory-speed tradeoff.
